airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/StateDecoratingIterator.java

```diff
@@ -186,20 +186,22 @@ protected final Optional<AirbyteMessage> getIntermediateMessage() {
    * read up so far
    *
    * @param isFinalState marker for if the final state of the iterator has been reached
+   * @param totalRecordCount count of read messages
    * @return AirbyteMessage which includes information on state of records read so far
    */
-  public AirbyteMessage createStateMessage(final boolean isFinalState) {
+  public AirbyteMessage createStateMessage(final boolean isFinalState, int totalRecordCount) {
     final AirbyteStateMessage stateMessage = stateManager.updateAndEmit(pair, currentMaxCursor, currentMaxCursorRecordCount);
     final Optional<CursorInfo> cursorInfo = stateManager.getCursorInfo(pair);
-    LOGGER.info("State report for stream {} - original: {} = {} (count {}) -> latest: {} = {} (count {})",
-        pair,
-        cursorInfo.map(CursorInfo::getOriginalCursorField).orElse(null),
-        cursorInfo.map(CursorInfo::getOriginalCursor).orElse(null),
-        cursorInfo.map(CursorInfo::getOriginalCursorRecordCount).orElse(null),
-        cursorInfo.map(CursorInfo::getCursorField).orElse(null),
-        cursorInfo.map(CursorInfo::getCursor).orElse(null),
-        cursorInfo.map(CursorInfo::getCursorRecordCount).orElse(null));
-
+    if (!isFinalState && totalRecordCount % 100 == 0) {
```

Comment:
why is the message only emitted if it is not a final state? Also, can you add a quick comment that we are logging once every 100 messages to reduce log verbosity

Reply:
> why is the message only emitted if it is not a final state? Also, can you add a quick comment that we are logging once every 100 messages to reduce log verbosity

removed `isFinalState` check and added comment